microsoft/AirSim
# Add ROS2 wrapper

Fixes: https://github.com/microsoft/AirSim/discussions/3830 


## About

Hi,

I found myself using AirSim & ROS but a lot of our logic and code is already implemented with ROS2.
I migrated the existing ROS wrapper to ROS2, with minimal changes as possible.

One major change is a new package contains all messages and services - `airsim_interfaces`.

**Nodes:**
- [x] airsim_node
- [x] pid_position_node 

**Packages:**
- [x] airsim_interfaces
- [x] airsim_ros_pkgs

Not all launch files were migrated, only the main so far - 
- [x] airsim_node.launch.py
- [x] static_transforms.launch.py
- [x] position_controller_simple.launch.py
- [x] airsim_with_simple_PD_position_controller.launch.py
- [x] dynamic_constraints.launch.py
- [x] rviz.launch.py

This PR probably need much more tests to check that everything work like ROS1 wrapper.
I thought it might be useful to share it with you guys, because ROS2 became very popular.

## How Has This Been Tested?

Test configuration:
- ROS2 Foxy/Galactic
- Ubuntu 20.04

It might work on different ROS2/Ubuntu versions as well.
Didn't check it.

Ran this ROS2 wrapper node.
Tested on one Multirotor and one Car with different sensors using ros2 command tool and Rviz2

## How to test it yourself?
- Install gcc >= 8.0.0:
` sudo apt-get install gcc-8 g++-8
  `Verify installation by `gcc-8 --version`
- install ROS2 pkgs:
 `sudo apt-get install ros-${ROS_DISTRO}-tf2-sensor-msgs ros-${ROS_DISTRO}-tf2-geometry-msgs ros-${ROS_DISTRO}-mavros* libyaml-cpp-dev`
- Verify you have colcon installed, if not - install it:
https://colcon.readthedocs.io/en/released/user/installation.html
- ` cd /AirSim/ros2`
- `colcon build --cmake-args -DCMAKE_C_COMPILER=gcc-8 --cmake-args -DCMAKE_CXX_COMPILER=g++-8 --cmake-args -DCMAKE_BUILD_TYPE=Debug` 
or for release `DCMAKE_BUILD_TYPE=Release`
- `source install/setup.bash`
- Run:
`ros2 launch airsim_ros_pkgs airsim_node.launch.py`
